contracts/Collection.sol

```diff
@@ -0,0 +1,123 @@
+        _setupRole(ROYALTY_MANAGER_ROLE, royaltyManager);
+    }
+
+    /** Sets Freeport contract address.
+     */
+    function setFreeport(address _freeportContract) public {
+        require(freeport == Freeport(address(0)));
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()));
+
+        freeport = Freeport(_freeportContract);
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId)
+    public view virtual override(BaseNFT) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// Issuer interface
+    /** Issue a supply of NFTs of a new type, and return its ID.
+     *
+     * No more NFT of this type can be issued again.
+     *
+     * The caller will be recorded as the issuer and it will initially own the entire supply.
+     */
+    function issue(uint64 supply, bytes memory data)
+    public returns (uint256) {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        return _issueAs(_msgSender(), supply, data);
+    }
+
+    /** Internal implementation of the function issue.
+     */
+    function _issueAs(address issuer, uint64 supply, bytes memory data)
+    internal returns (uint32) {
+        idCounter = idCounter + 1;
+
+        require(supply > 0);
+        _mint(issuer, idCounter, supply, data);
+
+        return idCounter;
+    }
+    /// Issuer interface
+
+    /// Royalty management interface
+    /** Configure the amounts and beneficiaries of royalties on primary and secondary transfers of this NFT.
+     *  Delegating to Freeport implementation.
+     */
+    function setupRoyaltyConfiguration(
+        uint32 innerNftId,
+        uint64 supply,
+        address primaryRoyaltyAccount,
+        uint256 primaryRoyaltyCut,
+        uint256 primaryRoyaltyMinimum,
+        address secondaryRoyaltyAccount,
+        uint256 secondaryRoyaltyCut,
+        uint256 secondaryRoyaltyMinimum)
+    public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender())
+            || hasRole(ROYALTY_MANAGER_ROLE, _msgSender()), "only admin or royalty manager");
+
+        freeport.configureRoyalties(
+            getGlobalNftId(innerNftId, supply),
+            primaryRoyaltyAccount,
+            primaryRoyaltyCut,
+            primaryRoyaltyMinimum,
+            secondaryRoyaltyAccount,
+            secondaryRoyaltyCut,
+            secondaryRoyaltyMinimum
+        );
+    }
+    /// Royalty management interface
+
+    /// NFT ID utilities
+    /** Calculate the global ID of an NFT type, identifying its inner nft id and supply.
+     */
+    function getGlobalNftId(uint32 innerNftId, uint64 supply)
+    public view returns (uint256) {
+        // issuer || innerNftId || supply: 160 + 32 + 64 = 256 bits
+        uint256 id = (uint256(uint160(address(this))) << (32 + 64))
+        | (uint256(innerNftId) << 64)
+        | uint256(supply);
```

Comment:
Why include the supply?

Reply:
I thought it would give us more flexibility. After all we call SC through the sdk so it must be fine in terms of security.
I will remove supply parameter.